File: 백준/Silver/10845. 큐/큐.py
from collections import deque
import sys
input = sys.stdin.readline  # 빠른 입력

q = deque()
n = int(input())

for _ in range(n):
    order = input().split()

    if order[0] == 'push':
        q.append(order[1])
    elif order[0] == 'pop':
        print(q.popleft() if q else -1)
    elif order[0] == 'size':
        print(len(q))
    elif order[0] == 'empty':
        print(0 if q else 1)
    elif order[0] == 'front':
        print(q[0] if q else -1)
    elif order[0] == 'back':
        print(q[-1] if q else -1)

File: 백준/Silver/10845. 큐/큐.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    queue<int> q;
    int n, x;
    string opr;
    cin >> n;

    while (n--) {
        cin >> opr;
        if (opr == "push") {
            cin >> x;
            q.push(x);
        }
        else if (opr == "pop") {
            if (q.empty()) cout << -1 << '\n';
            else {
                cout << q.front() << '\n';
                q.pop();
            }
        }
        else if (opr == "size") cout << q.size() << '\n';
        else if (opr == "empty") {
            if (q.empty()) cout << 1 << '\n';
            else cout << 0 << '\n';
        }
        else if (opr == "front") {
            if (q.empty()) cout << -1 << '\n';
            else cout << q.front() << '\n';
        }
        else {
            if (q.empty()) cout << -1 << '\n';
            else cout << q.back() << '\n';
        }
    }
}
